tests/gtestmusig2.c

```diff
@@ -0,0 +1,416 @@
+    ASSERT_EQ(err, 1);
+
+    /**** Verification ****/
+    /* Verify the aggregated signature with secp256k1_schnorrsig_verify */
+    err = musig2_ver_musig(ctx, signature1, mca1.aggr_pk, MSG_1, MSG_1_LEN);
+    ASSERT_EQ(err, 0);
+
+}
+
+TEST (musig2, non_corresponding_signers) {
+
+    secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+
+
+    int i, j, k, l;
+    int ind;
+    int nr_participants = NR_SIGNERS + 1;
+    secp256k1_pubkey pk_list[nr_participants];    // Signers' public key list
+    secp256k1_pubkey batch_list[nr_participants * V * NR_MESSAGES];   // Stores the batches of signers
+    musig2_context_sig mcs_list[nr_participants]; // Array that holds NR_SIGNERS musig2_context_sig
+
+    int err;
+
+
+    /**** Initialization ****/
+    for (i = 0; i < nr_participants; i++) {
+        /* Generate a keypair for the signer and get batch commitments. */
+        err = musig2_init_signer(&mcs_list[i], ctx, NR_MESSAGES);
+        ASSERT_EQ(err, 1);
+
+        /* Store the public key of the signer in pk_list */
+        err = secp256k1_keypair_pub(ctx, &pk_list[i], &mcs_list[i].keypair);
+        ASSERT_EQ(err, 1);
+
+        /* Store the batch commitments of the signer in batch_list */
+        l = 0; // the index of the signer's commitment list.
+        for (k = 0; k < NR_MESSAGES; k++) {
+            for (j = 0; j < V; j++, l++) {
+                ind = (nr_participants * V * k) + (j * nr_participants) + i; // the index for the list of collected batches.
+                err = secp256k1_keypair_pub(ctx, &batch_list[ind], mcs_list[i].comm_list[l]);
+                ASSERT_EQ(err, 1);
+            }
+        }
+    }
+
+    /**** Aggregate the public keys and batch commitments for each signer ****/
+    int cnt = 0;
+    for (i = 1; i < nr_participants; i++)
+        cnt += musig2_aggregate_pubkey(mcs_list[i].mc, pk_list, NR_SIGNERS);
+    ASSERT_EQ(cnt, NR_SIGNERS);
+
+    cnt = 0;
+    for (i = 0; i < NR_SIGNERS; i++)
+        cnt += musig2_aggregate_R(&mcs_list[i + 1], batch_list);
+    ASSERT_EQ(cnt, NR_SIGNERS);
+
+
+    /**** Signature ****/
+    musig2_partial_signatures mps1[NR_SIGNERS];
+
+    for (i = 0; i < NR_SIGNERS; i++) {
+        /* Generate the partial signatures */
+        err = musig2_sign(&mcs_list[i + 1], MSG_1, MSG_1_LEN, mps1[i].sig);
+        ASSERT_EQ(err, 1);
+        memcpy(mps1[i].R.data, mcs_list[i + 1].mc->aggr_R.data, PK_BYTES);
+    }
+
+    /**** Aggregation ****/
+    musig2_context mca1;
+    unsigned char signature1[SCH_SIG_BYTES];
+    err = musig2_aggregate_partial_sig(ctx, &mca1, mps1, pk_list, signature1, NR_SIGNERS);
+    ASSERT_EQ(err, 1);
+
+    /**** Verification ****/
+    /* Verify the aggregated signature with secp256k1_schnorrsig_verify */
+    err = musig2_ver_musig(ctx, signature1, mca1.aggr_pk, MSG_1, MSG_1_LEN);
+    ASSERT_EQ(err, 0);
+
+}
+
+TEST (musig2, incorrect_nonce) {
```

Comment:
This test seems to check the incorrect aggregated commitment of nonces, rather than the incorrect nonce. We should change the name of the test accordingly.